File: src/packets/PacketC2SSourceControl.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x32
 * Source Control
 * Control source media
 * Data:
 *      (Unsigned Char) Source ID
 *      (Unsigned Char) Button ID
 */
class PacketC2SSourceControl extends PacketC2S {
    parseData() {
        this._sourceID = this._buffer.readUInt8();
        this._operation = this._buffer.readUInt8();
    }

    getSourceID() {
        return this._sourceID;
    }

    getOperation() {
        return this._operation;
    }

    getID() {
        return PacketC2SSourceControl.ID;
    }
}

PacketC2SSourceControl.ID = 0x32;

module.exports = PacketC2SSourceControl;

File: src/RNetPi.Core/Packets/PacketC2SSourceControl.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x32
/// Source Control
/// Control source media
/// Data:
///     (Unsigned Char) Source ID
///     (Unsigned Char) Button ID
/// </summary>
public class PacketC2SSourceControl : PacketC2S
{
    public const byte ID = 0x32;

    public byte SourceID { get; private set; }
    public byte Operation { get; private set; }

    public PacketC2SSourceControl(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        SourceID = Reader.ReadByte();
        Operation = Reader.ReadByte();
    }
}